cmd/medius/images/verify.go

```diff
@@ -0,0 +1,197 @@
+package images
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/rand"
+	"errors"
+	"path"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	urand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+	v1 "kubevirt.io/api/core/v1"
+	kvirtcli "kubevirt.io/client-go/kubecli"
+	kvirtlog "kubevirt.io/client-go/log"
+	"kubevirt.io/containerdisks/cmd/medius/common"
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/docs"
+)
+
+const (
+	VerifyUsername = "verify"
+)
+
+func NewVerifyImagesCommand(options *common.Options) *cobra.Command {
+	options.VerifyImagesOptions = common.VerifyImageOptions{
+		Namespace: "kubevirt",
+		Timeout:   600,
+	}
+
+	verifyCmd := &cobra.Command{
+		Use:   "verify",
+		Short: "Verify that containerdisks are bootable and guests are working",
+		Run: func(cmd *cobra.Command, args []string) {
+			results, err := readResultsFile(options.ResultsFile)
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			// Silence the kubevirt client log
+			kvirtlog.Log = kvirtlog.MakeLogger(kvirtlog.NullLogger{})
+			client, err := kvirtcli.GetKubevirtClient()
+			if err != nil {
+				logrus.Fatal(err)
+			}
+
+			err = spawnWorkers(cmd.Context(), options, func(a api.Artifact) error {
+				result, ok := results[a.Metadata().Describe()]
```

Comment:
Here I also would love to see this function to get the "result" from the push step getting passed in and returning the modified result, which can then be collected via a channel from the `spawnWorkers` context.